src/client/Client.js

```diff
@@ -402,10 +408,26 @@ class Client extends BaseClient {
 
     const query = new URLSearchParams({
       client_id: this.application.id,
-      scope: 'bot',
     });
 
-    if (options.permissions) {
+    const scopes = options.scopes;
+    if (typeof scopes === 'undefined') {
+      throw new TypeError('INVITE_MISSING_SCOPES');
+    }
+    if (!Array.isArray(scopes)) {
+      throw new TypeError('INVALID_TYPE', 'scopes', 'Array of Invite Scopes', true);
+    }
+    const requiredScope = scopes.find(scope => scope === 'bot' || scope === 'applications.commands');
+    if (!requiredScope) {
+      throw new TypeError('INVITE_MISSING_SCOPES');
+    }
+    const invalidScope = scopes.find(scope => !InviteScopes.includes(scope));
+    if (invalidScope) {
+      throw new TypeError('INVALID_ELEMENT', 'Array', 'scopes', invalidScope);
+    }
+    query.set('scope', [...scopes].join(' '));
```

Comment:
Unless you're supporting enumerators, which would make it throw in the previous lines, this isn't necessary:
```suggestion
    query.set('scope', scopes.join(' '));
```